src/lib/QtWidgets/QTabWidget/index.ts

```diff
@@ -0,0 +1,43 @@
+import addon from "../../core/addon";
+import { NodeWidget } from "../QWidget";
+import { BaseWidgetEvents } from "../../core/EventWidget";
+import { NativeElement } from "../../core/Component";
+import { QIcon } from "../../QtGui/QIcon";
+import { TabPosition } from "../../QtEnums";
+
+export const QTabWidgetEvents = Object.freeze({
+  ...BaseWidgetEvents,
+  currentChanged: "currentChanged",
+  tabBarClicked: "tabBarClicked",
+  tabBarDoubleClicked: "tabBarDoubleClicked",
+  tabCloseRequested: "tabCloseRequested"
+});
+
+export class QTabWidget extends NodeWidget {
+  native: NativeElement;
+  constructor(parent?: NodeWidget) {
+    let native;
+    if (parent) {
+      native = new addon.QTabWidget(parent.native);
+    } else {
+      native = new addon.QTabWidget();
+    }
+    super(native);
+    this.parent = parent;
+    this.native = native;
+    // bind member functions
+    this.addTab.bind(this);
+  }
+
+  addTab(page: NodeWidget, icon: QIcon, label: string) {
```

Comment:
The GC thing is happening because we have sent the widget to native instance but we didnt have any reference of tab in the javascript world anymore. So I would recommend adding a childTabs property to QTabWidget which would store the javascript reference as well. For more details you can take a look at addWidget's implementation in the flex layout. I am storing children in a set there.  If you need more clearer code example let me know.

Reply:
I've even looked at the implementation but was confused by the Yoga nodes so was sure its related to that implementation. Its working as intended now 👍